### PR TITLE
Adapt .editorconfig for PHPStorm

### PR DESCRIPTION
**Pull request by nupplaphil**
@hankg this is for you :-)

Works now like a charm for most of our files

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -6,9 +6,13 @@ root = true
 [*]
 charset = utf-8
 end_of_line = lf
-trim_trailing_whitespaces = true
 indent_style = tab
-quote_type = single
```
Comment (MrPetovan) [inline .editorconfig:11]: Why did you remove this?

Comment (nupplaphil) [reply .editorconfig:11]: It's not officially supported: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#ideas-for-domain-specific-properties

**Comment by HankG** [comment]
> @HankG this is for you :-)
> 
> Works now like a charm for most of our files

Does it automatically pick it up on the fly or should I blow away the .idea folder etc. and reimport/open the project? For the record I've blown away all system settings in ~/.local and ~/.var to figure out WTH has gone sideways from me trying to fix things before. I think that should be sufficient to reset it to factory settings.

File: .editorconfig (hunk 1 above)
Comment (nupplaphil) [reply .editorconfig:11]: OK, reverted it, because **maybe** some IDEs use it (I'm not aware of it *g*)

**Review by MrPetovan — COMMENTED** [review]
> Works now like a charm for most of our files

What about the `*.md` files in the `doc/` folder? @HankG had issues with these.

**Comment by nupplaphil** [comment]
It was enough to just clone the repository and use PHPStorm.
This is how it looks like in the settings:
image
And this is my footer-bar:
image

**Comment by nupplaphil** [comment]
> > Works now like a charm for most of our files
> 
> What about the `*.md` files in the `doc/` folder? @HankG had issues with these.

puh, I see what you mean.. the result of executing the auto-formatter for  `*.md` files is disastrous

**Comment by MrPetovan** [comment]
So do you have more work to add to this PR for the Markdown files?

**Comment by nupplaphil** [comment]
I think it isn't possible to set a different codeblock identation in a Markdown file ... at least not with `.editorconfig`, I'm currently searching some other references

So yes, I think you can go ahead merging it :-/

**Comment by nupplaphil** [comment]
> > @HankG this is for you :-)
> > Works now like a charm for most of our files
> 
> Does it automatically pick it up on the fly or should I blow away the .idea folder etc. and reimport/open the project? For the record I've blown away all system settings in ~/.local and ~/.var to figure out WTH has gone sideways from me trying to fix things before. I think that should be sufficient to reset it to factory settings.

I didn't change anything the last time I cloned the Friendica codebase. I works the way it was predefined ..

**Comment by HankG** [comment]
> > > @HankG this is for you :-)
> > > Works now like a charm for most of our files
> > 
> > 
> > Does it automatically pick it up on the fly or should I blow away the .idea folder etc. and reimport/open the project? For the record I've blown away all system settings in ~/.local and ~/.var to figure out WTH has gone sideways from me trying to fix things before. I think that should be sufficient to reset it to factory settings.
> 
> I didn't change anything the last time I cloned the Friendica codebase. I works the way it was predefined ..

K. I think I'll blow away my .idea folder as well after this is merged and try again.

**Comment by nupplaphil** [comment]
@MrPetovan / @HankG  - I now added my whole PHPStorm code style XML

Maybe we can tweak it as we're currently using it and add the minimum, we all agree to :)
It seems like PHPStorm exports a diff XML to the default settings, which makes it lightweight

**Comment by MrPetovan** [comment]
Can you please confirm editing `*.md` files doesn't yield the same mixed indentation @HankG faced in his previous PR?

**Comment by nupplaphil** [comment]
> Can you please confirm editing `*.md` files doesn't yield the same mixed indentation @HankG faced in his previous PR?

I cannot confirm this :-/
- All `*.md` files are using the (PHP like), global `[*]` setting
- Reformatting the Markdown files automatically adds tabs everywhere it can (including all Codeblocks)
  - Unfortunately, it doesn't switch the indentation based on the code block language
- In some cases, where tabs aren't possible, it adds whitespaces

So unfortunately, it's as disastrous, as it is for @HankG  ..

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Ah, well.

**Comment by HankG** [comment]
I blew away the .idea file and all the ~/.var and ~/.local files but it still is doing the settings whichever way it wants to. I'm going to follow your screenshot above to fix it but not sure why it isn't honoring this config file when bootstrapping the project :(.

Screenshot from 2023-02-27 20-11-23
Screenshot from 2023-02-27 20-10-58
